### ISSUE TITLE
Fails to build on iOS - script shell error w/ exit code 126

### ISSUE THREAD

**Issue by DaveVoyles** [manifoldjs/ManifoldJS#100]
**Steps to reproduce:**
1. Use Manifold web page to build a project on Windows (Will no work on OS X, [see my other issue](https://github.com/manifoldjs/ManifoldJS/issues/101))
2.  Open and attempt to run the Cordova project on iOS 

**Error:**
![](https://dl.dropboxusercontent.com/s/3il6ywkpxowns87/Screenshot%202015-06-25%2017.07.47.png?dl=0raw=1)

**Comment by f2bo** [comment]
This happens when you copy the project files generated by Windows to the Mac. The executable bit is lost on some scripts. We'll look into this a bit more to see if there's something ManifoldJS can do to help with this but, for the time being, the manual workaround is to restore the executable bit on the scripts. At the very least you need to do this for **cordova/lib/copy-www-build-step.sh**, though there are other Cordova scripts that may also need fixing.

```
chmod +x [path to the ManifoldJS project]/cordova/platforms/ios/cordova/lib/copy-www-build-step.sh
```
We tested this on a project generated on Windows and then copied to the Mac where we ran ManifoldJS against the FamousAngularPokemonAzurewebsitesNet site that you appear to be using. After setting the executable bit on the copy-www-build-step.sh script, we successfully built and launched the app in the iOS Simulator.

**Comment by boyofgreen** [comment]
Dave, can you validate this fix for us?  It's not idea, but the only way we can think to make the code portable from PC to mac.  If its not portable, there is no point in building it :|

**Comment by DaveVoyles** [comment]
Sure thing -- I'll be back at my desk tomorrow, so I'll give this a go and report back with the update.

**Comment by f2bo** [comment]
We implemented a solution that restores the executable permission when the project is built in OS X. The fix is in the Hosted Web App plugin and is implemented as a hook that executes during the prepare phase. 

It's still not ideal because you need to open a Terminal window and run `cordova prepare` at least once in the cordova root folder before you load the project in Xcode.

Note that this fix has not been published yet. If you are feeling adventurous and want to test it, you'll need to remove the current _cordova-plugin-hostedwebapp_ plugin from the project generated by ManifoldJS and and then clone the ManifoldCordova repository locally, and reinstall the plugin from the v0.1.2-fix-ios-permissions branch.

**Comment by DaveVoyles** [comment]
OK I'm going to try this out. To be clear, I need to do the following:

1) Generate the project via Manifold on a PC
2) Bring it over to OS X

Not sure about the next step though:

"where we ran ManifoldJS against the FamousAngularPokemonAzurewebsitesNet site that you appear to be using. After setting the executable bit on the copy-www-build-step.sh script, we successfully built and launched the app in the iOS Simulator."

Am I running a manifold command again on OS X and ( from this v0.1.2-fix-ios-permissions branch) and copying the step.sh script into the OS X project?

**Comment by f2bo** [comment]
No. Sorry I wasn't clear. You don't need to generate the app with ManifoldJS again.

You have two options here using the project that you generated previously in Windows and copied to the Mac.

a) **Completely manual step:** In OS X, you can open a Terminal window and execute:
```
chmod +x [path to the project]/cordova/platforms/ios/cordova/lib/copy-www-build-step.sh
```
b) **Semi-automated step:** In OS X, uninstall the cordova-plugin-hostedwebapp plugin and reinstall it from the _v0.1.2-fix-ios-permissions branch_.
- Go to [path to the project]/cordova
- In a Terminal window run
  ```
  cordova plugin rm cordova-plugin-hostedwebapp  

  cordova plugin add https://github.com/manifoldjs/ManifoldCordova#v0.1.2-fix-ios-permissions  

  cordova prepare  
  ```

> Note: I'm exaggerating by calling this step "semi-automated" because by the time the fix is incorporated into ManifoldCordova and the plugin republished, ManifoldJS will start using it and you won't need to remove and reinstall the plugin. You will only call **cordova prepare** at least once, which you would probably do anyway. 

**WARNING:** If the package you bring over from Windows to the Mac includes the Windows platform, running cordova prepare will fail. This is a problem in the current Cordova release. You need to remove the Windows platform in OS X.

```
cordova platform rm windows
```

**Comment by DaveVoyles** [comment]
My path for OS X is slightly different, but definitely missing some of those files at the end, so I'm not able to take that approach. 

**Step 1** This morning I created the Manifold project from the web tool within a windows machine. I then copied that .zip over to my OS X machine. Should I have done it via comand line instead?

![img ] (dl.dropboxusercontent.com/s/zdnkw4wipr0qfrb/Screenshot%202015-07-08%2017.54.47.png?dl=0)

**Step 2**
Oddly enough, the *cordova prepare* command did not fail, despite me having forgotten to delete the Windows 10 build.

I clicked on the Xcode project in the projects/cordova/platforms/ios folder and it worked just fine.

**Comment by f2bo** [comment]
Hi Dave,

Thanks for testing this out.

The scripts folder is not CordovaLib but cordova/lib (the one above CordovaLib in the screenshot). Yes, I know. I didn't choose the names. :)

I need to double check, but I suspect that if you used the web tool and then copied the .zip to the OS X machine, then you wouldn't hit the same issue that you reported previously and the build would have worked without any additional steps. 

At least, that's my understanding of the problem until now. We did reproduce the same error you reported by copying individual files generated by ManifoldJS on a Windows machine to the Mac -- not a .zip archive. However, I just tested generating a package with the web tool from a Mac and it built without problems. 

I don't have the right environment available to do this right now, so I didn't generate the package from a Windows machine and then copied the .zip to the Mac as you did. I will test it anyway as soon as I can. Nevertheless, I would be surprised if that failed.

Regarding the _cordova prepare_ **not** failing, it just dawned on me that I neglected to consider that you might have a different version of Cordova installed globally and perhaps that one doesn't have the same problem. ManifoldJS is built against a specific version of Cordova, which it installs privately. However, when you run cordova prepare from a Terminal window, you are actually using the globally installed version. We need to look into this particular scenario a bit more to evaluate the implications. 

In any case, can I ask you to run "cordova -v" from a Terminal window to find out which version is installed globally. 

Thanks!

**Comment by DaveVoyles** [comment]
This one is brief -- running out the door. Great explanation, though! 

Yes, the folder naming scheme can definitely be confusing at times :)

I am running Cordova 5.1.1 on OS X.

**Comment by f2bo** [comment]
We decided not to update the plugin to do the "semi-automatic" step. We think this is not such a common scenario. Instead, we'll update the "Known Issues" section of the Readme document with instructions on how to fix the permissions manually (see 0b992648931397c0f4864501e914d032fed90801)
